frontend/src/components/ViewTrips/add-trip.js

```diff
@@ -0,0 +1,143 @@
+                                    placeholder, ref, subFormText = '') {
+  let formControl;
+  if (inputType === 'text') {
+    formControl = createTextFormControl(placeholder, ref)
+  } else {
+    // TODO(Issue #52): Create diff form inputs for start & end dates
+    //                  and collaborator emails.
+    console.error('Only text form inputs are implemented as of now');
+  }
+
+  return (
+    <Form.Group controlId={controlId}>
+      <Form.Label>{formLabel}</Form.Label>
+        {formControl}
+      {/* Temporary instructions until fix Issue #52 */}
+      <Form.Text className="text-muted">
+        {subFormText}
+      </Form.Text>
+    </Form.Group>
+  )
+}
+
+/**
+ * Component corresponding to add trips modal.
+ *
+ * @param {Object} props These are the props for this component:
+ * - show: Boolean that determines if the add trips modal should be displayed.
+ * - handleClose: The function that handles closing the add trips modal.
+ * - userEmail: The current users email.
+ *
+ * @extends React.Component
+ */
+class AddTrip extends React.Component {
+  /** @inheritdoc */
+  constructor(props) {
+    super(props);
+
+    this.nameRef = React.createRef();
+    this.descriptionRef = React.createRef();
+    this.destinationRef = React.createRef();
+    this.startDateRef = React.createRef();
+    this.endDateRef = React.createRef();
+    this.collaboratorsRef = React.createRef();
+  }
+
+  /**
+   * Upon submission of the form, a new Trip document is created and the add
+   * trip modal is closed.
+   */
+  submitNewTrip = () => {
+    console.log('New trip submitted.');
+    createTrip({name: this.nameRef.current.value,
+               description: this.descriptionRef.current.value,
+               destination: this.destinationRef.current.value,
+               startDate: this.startDateRef.current.value,
+               endDate: this.endDateRef.current.value,
+               collaborators: this.collaboratorsRef.current.value
+    });
+
+    this.props.handleClose();
+  }
+
+  /** @inheritdoc */
+  render() {
+    return (
+      <Modal show={this.props.show} onHide={this.props.handleClose}>
+        <Modal.Header closeButton>
+          <Modal.Title>Add New Trip</Modal.Title>
+        </Modal.Header>
+
+        <Form>
+          <Modal.Body>
+            {createFormGroup('tripNameGroup', 'Trip Name', 'text',
+                             'Enter Trip Name', this.nameRef)}
+            {createFormGroup('tripDescripGroup', 'Trip Description', 'text',
+                             'Enter Trip Description', this.descriptionRef)}
+            {createFormGroup('tripDestGroup', 'Trip Destination', 'text',
+                             'Enter Trip Destination', this.destinationRef)}
+            {createFormGroup('tripStartDateGroup', 'Start Date', 'text',
+                            'Enter Trip Start Date', this.startDateRef,
+                            'Enter date in the form: \'mm/dd/yyy\'')}
```

Comment:
Is it intentional that there are only 3 y's here and below?

Reply:
Oops yes there should be 4 y's. Fixed.